src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -911,7 +914,7 @@ private AuthenticationTicket ValidateToken(string idToken, OpenIdConnectMessage
                 protocolValidationContext.AuthorizationCode = message.Code;
```

Comment:
I set the Authorization Code to null to skip the chash validation. If a code is present as ValidationContext.AuthorizationCode in old model or ValidationContext.Message.Code in new model, protocolValidator will try to validate the chash and throw if it cannot find the chash claim which is the case in 'code' only flow.
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/KDev/src/Microsoft.IdentityModel.Protocols.OpenIdConnect/OpenIdConnectProtocolValidator.cs#L378

Ideal way to fix this is to have the protocolValidator look at the state and if response_type is 'code' in the request sent out, do not validate the chash.